Installation issue: gsoap missing autoheader

### Steps to reproduce the issue

```console
$ docker run --rm -it spack/ubuntu-noble:develop
Unable to find image 'spack/ubuntu-noble:develop' locally
develop: Pulling from spack/ubuntu-noble
...
Digest: sha256:2d087ba3fde826b2cc3fb628b8411b965726c4b1f0796f6e2175cbb140f7b360
Status: Downloaded newer image for spack/ubuntu-noble:develop
root@3f6c86fff107:~# source /opt/spack/share/spack/setup-env.sh
root@3f6c86fff107:~# spack install gsoap
...
==> Installing gsoap-2.8.135-5xfcxnizt5x7opxcah6w25hbbtj5d7f7 [27/27]
==> No binary for gsoap-2.8.135-5xfcxnizt5x7opxcah6w25hbbtj5d7f7 found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/b1/b11757e405d55d4674dfbf88c4fa6d7e24155cf64ed8ed578ccad2f2b555e98d.zip
==> No patches needed for gsoap
==> gsoap: Executing phase: 'autoreconf'
==> gsoap: Executing phase: 'configure'
==> gsoap: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    '/opt/spack/opt/spack/linux-ubuntu24.04-skylake/gcc-13.3.0/gmake-4.4.1-nqkodtnx36hwt2ovsasbibdfpmc3hhtr/bin/make' '-j8' 'V=1'

2 errors found in build log:
     207    ==> [2024-12-20-16:31:53.195233] FILTER FILE: /tmp/root/spack-stage/spack-stage-gsoap-2.8.135-5xfcxniz
            t5x7opxcah6w25hbbtj5d7f7/spack-src/gsoap/mod_gsoap/mod_gsoap-1.0/apache_13/libtool [replacing "^pic_fl
            ag=""$"]
     208    ==> [2024-12-20-16:31:53.198842] FILTER FILE: /tmp/root/spack-stage/spack-stage-gsoap-2.8.135-5xfcxniz
            t5x7opxcah6w25hbbtj5d7f7/spack-src/gsoap/mod_gsoap/mod_gsoap-1.0/apache_13/libtool [replacing "^pic_fl
            ag=""$"]
     209    ==> gsoap: Executing phase: 'build'
     210    ==> [2024-12-20-16:31:53.246839] '/opt/spack/opt/spack/linux-ubuntu24.04-skylake/gcc-13.3.0/gmake-4.4.
            1-nqkodtnx36hwt2ovsasbibdfpmc3hhtr/bin/make' '-j8' 'V=1'
     211    (CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/bash '/tmp/root/spack-stage/spack-stage-gsoap-2.8.135-5xfc
            xnizt5x7opxcah6w25hbbtj5d7f7/spack-src/missing' autoheader)
     212    /tmp/root/spack-stage/spack-stage-gsoap-2.8.135-5xfcxnizt5x7opxcah6w25hbbtj5d7f7/spack-src/missing: li
            ne 81: autoheader: command not found
  >> 213    WARNING: 'autoheader' is missing on your system.
     214             You should only need it if you modified 'acconfig.h' or
     215             'configure.ac' or m4 files included by 'configure.ac'.
     216             The 'autoheader' program is part of the GNU Autoconf package:
     217             <https://www.gnu.org/software/autoconf/>
     218             It also requires GNU m4 and Perl in order to run:
     219             <https://www.gnu.org/software/m4/>
     220             <https://www.perl.org/>
  >> 221    make: *** [Makefile:426: config.h.in] Error 127

See build log for details:
  /tmp/root/spack-stage/spack-stage-gsoap-2.8.135-5xfcxnizt5x7opxcah6w25hbbtj5d7f7/spack-build-out.txt
```

### Error message

<details><summary>Error message</summary>
<pre>
==> Installing gsoap-2.8.135-5xfcxnizt5x7opxcah6w25hbbtj5d7f7 [27/27]
==> No binary for gsoap-2.8.135-5xfcxnizt5x7opxcah6w25hbbtj5d7f7 found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/b1/b11757e405d55d4674dfbf88c4fa6d7e24155cf64ed8ed578ccad2f2b555e98d.zip
==> No patches needed for gsoap
==> gsoap: Executing phase: 'autoreconf'
==> gsoap: Executing phase: 'configure'
==> gsoap: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    '/opt/spack/opt/spack/linux-ubuntu24.04-skylake/gcc-13.3.0/gmake-4.4.1-nqkodtnx36hwt2ovsasbibdfpmc3hhtr/bin/make' '-j8' 'V=1'

2 errors found in build log:
     207    ==> [2024-12-20-16:31:53.195233] FILTER FILE: /tmp/root/spack-stage/spack-stage-gsoap-2.8.135-5xfcxniz
            t5x7opxcah6w25hbbtj5d7f7/spack-src/gsoap/mod_gsoap/mod_gsoap-1.0/apache_13/libtool [replacing "^pic_fl
            ag=""$"]
     208    ==> [2024-12-20-16:31:53.198842] FILTER FILE: /tmp/root/spack-stage/spack-stage-gsoap-2.8.135-5xfcxniz
            t5x7opxcah6w25hbbtj5d7f7/spack-src/gsoap/mod_gsoap/mod_gsoap-1.0/apache_13/libtool [replacing "^pic_fl
            ag=""$"]
     209    ==> gsoap: Executing phase: 'build'
     210    ==> [2024-12-20-16:31:53.246839] '/opt/spack/opt/spack/linux-ubuntu24.04-skylake/gcc-13.3.0/gmake-4.4.
            1-nqkodtnx36hwt2ovsasbibdfpmc3hhtr/bin/make' '-j8' 'V=1'
     211    (CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/bash '/tmp/root/spack-stage/spack-stage-gsoap-2.8.135-5xfc
            xnizt5x7opxcah6w25hbbtj5d7f7/spack-src/missing' autoheader)
     212    /tmp/root/spack-stage/spack-stage-gsoap-2.8.135-5xfcxnizt5x7opxcah6w25hbbtj5d7f7/spack-src/missing: li
            ne 81: autoheader: command not found
  >> 213    WARNING: 'autoheader' is missing on your system.
     214             You should only need it if you modified 'acconfig.h' or
     215             'configure.ac' or m4 files included by 'configure.ac'.
     216             The 'autoheader' program is part of the GNU Autoconf package:
     217             <https://www.gnu.org/software/autoconf/>
     218             It also requires GNU m4 and Perl in order to run:
     219             <https://www.gnu.org/software/m4/>
     220             <https://www.perl.org/>
  >> 221    make: *** [Makefile:426: config.h.in] Error 127

See build log for details:
  /tmp/root/spack-stage/spack-stage-gsoap-2.8.135-5xfcxnizt5x7opxcah6w25hbbtj5d7f7/spack-build-out.txt

</pre></details>


### Information on your system

* **Spack:** 0.24.0.dev0 (6de1ebd71a05d28c04e2ad5c0ec1755e3ced9136)
* **Python:** 3.12.3
* **Platform:** linux-ubuntu24.04-skylake


### Additional information

[spack-build-out.txt](https://github.com/user-attachments/files/18213086/spack-build-out.txt)
[spack-build-env.txt](https://github.com/user-attachments/files/18213087/spack-build-env.txt)

Maintainers: @gartung  @greenc-FNAL  @marcmengel  @vitodb

I wonder if this is due to the changing of files after configure, in, e.g.:
https://github.com/spack/spack/blob/2edbed3a9db73efed01c37037cb2bfd499f5b569/lib/spack/spack/build_systems/autotools.py#L395-L419

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/gsoap/package.py
+++ b/var/spack/repos/builtin/packages/gsoap/package.py
@@ -46,6 +46,10 @@ class Gsoap(AutotoolsPackage, SourceforgePackage):
     depends_on("pkgconfig", type="build")
     depends_on("bison", type="build")
     depends_on("flex", type="build")
+    depends_on("autoconf", type="build")
+    depends_on("automake", type="build")
+    depends_on("libtool", type="build")
+    depends_on("m4", type="build")
 
     def configure_args(self):
         return ["--enable-ipv6"]

